[NP-1834] Add CrunchService maybeIntercept method

KernelDeimos opened the pull request:
Adds a method any service can use to require a capability or an option from an array of capabilities.

Example:
```
CrunchService crunchService = (CrunchService) x.get("crunchService");
crunchService.require(new String[]{"554af38a-8225-87c8-dfdf-eeb15f71215f-49"});
```

Commit:
Add CrunchService require method

kgrgreer commented:
What does it mean to "require" a Capability? Could you use a different term to not overload FOAM's current use of "require"? Please provide more docs/comments.

Commit:
Rename CrunchService.require to maybeIntercept and add comments

Commit:
Update logger message in CrunchService.maybeIntercept

KernelDeimos commented:
Updated to `maybeIntercept` as a possibly better name and added some docs. This method allows a block of code to throw CapabilityRuntimeException if a capability isn't satisfied, without necessarily creating a permission.
